packages/features/bookings/lib/conflictChecker/checkForConflicts.ts

```diff
@@ -1,45 +1,59 @@
-const log = logger.getSubLogger({ prefix: ["[api] book:user"] });
 
 // if true, there are conflicts.
-export function checkForConflicts(busyTimes: BufferedBusyTimes, time: dayjs.ConfigType, length: number) {
+export function checkForConflicts({
+  busy,
+  time,
+  eventLength,
+  currentSeats,
+}: {
+  busy: BufferedBusyTimes;
+  time: Dayjs;
+  eventLength: number;
+  currentSeats?: CurrentSeats;
+}) {
   // Early return
-  if (!Array.isArray(busyTimes) || busyTimes.length < 1) {
+  if (!Array.isArray(busy) || busy.length < 1) {
     return false; // guaranteed no conflicts when there is no busy times.
   }
-
-  for (const busyTime of busyTimes) {
-    const startTime = dayjs(busyTime.start);
-    const endTime = dayjs(busyTime.end);
-    // Check if time is between start and end times
-    if (dayjs(time).isBetween(startTime, endTime, null, "[)")) {
-      log.error(
-        `NAUF: start between a busy time slot ${safeStringify({
-          ...busyTime,
-          time: dayjs(time).format(),
-        })}`
-      );
+  // no conflicts if some seats are found for the current time slot
+  if (currentSeats?.some((booking) => booking.startTime.toISOString() === time.toISOString())) {
+    return false;
+  }
+  const slotStartDate = time.utc().toDate();
+  const slotEndDate = time.add(eventLength, "minutes").utc().toDate();
+  // handle busy times.
+  return busy.some((busyTime) => {
+    const busyStartDate = dayjs.utc(busyTime.start).toDate();
+    const busyEndDate = dayjs.utc(busyTime.end).toDate();
+    // First check if there's any overlap at all
+    // If busy period ends before slot starts or starts after slot ends, there's no overlap
+    if (busyEndDate <= slotStartDate || busyStartDate >= slotEndDate) {
+      return false;
+    }
+    // Now check all possible overlap scenarios:
+    // 1. Slot start falls within busy period (inclusive start, exclusive end)
+    if (slotStartDate >= busyStartDate && slotStartDate < busyEndDate) {
       return true;
     }
-    // Check if slot end time is between start and end time
-    if (dayjs(time).add(length, "minutes").isBetween(startTime, endTime)) {
-      log.error(
-        `NAUF: Ends between a busy time slot ${safeStringify({
-          ...busyTime,
-          time: dayjs(time).add(length, "minutes").format(),
-        })}`
-      );
+    // 2. Slot end falls within busy period (exclusive start, inclusive end)
+    if (slotEndDate > busyStartDate && slotEndDate <= busyEndDate) {
       return true;
     }
-    // Check if startTime is between slot
-    if (startTime.isBetween(dayjs(time), dayjs(time).add(length, "minutes"))) {
+    // 3. Busy period completely contained within slot
+    if (busyStartDate >= slotStartDate && busyEndDate <= slotEndDate) {
       return true;
     }
-  }
+    // 4. Slot completely contained within busy period
+    if (busyStartDate <= slotStartDate && busyEndDate >= slotEndDate) {
+      return true;
+    }
+  });
   return false;
```

Comment:
This false return statement is un reachable

Reply:
Fixed 👍